src/Mod/PartDesign/Gui/TaskPadParameters.cpp

```diff
@@ -104,6 +104,11 @@ void TaskPadParameters::onModeChanged(int index)
         break;
     case Modes::ToFace:
         pcPad->Type.setValue("UpToFace");
+        // check button if there is no face yet
+        if (ui->lineFaceName->text().isEmpty()) {
+            ui->buttonFace->setChecked(true);

```

Comment:
I suppose this is your main change? If so that's reasonable. As such the button an the text box should work in tandem I believe: while the box is focused the button should be checked and vice versa.

Reply:
Yes. This is just a logic thing to guide the user.

Reply:
In Pad and Pocket you have the exact same code block. The result is that the identical string for translation appears 3x in the .ts files. To fix this and the code duplication move this block to a protected function in TaskExtrudeParameters and call it here (and in Pocket).

Reply:
Thanks. Done now.